### ISSUE TITLE
Add documentation for new SuppressMessageAttribute scope

### ISSUE THREAD

**Issue by mavasani** [dotnet/docs#9046]
https://github.com/dotnet/roslyn/pull/31092 adds a new SuppressMessageAttribute scope for very widely requested feature: https://github.com/dotnet/roslyn/issues/486. We should document it here (and also improve the documentation for current scopes) once the fix goes out - Dev16 Preview2.

Also note that the new scope is only valid for Roslyn analyzers, and is not going to be respected for legacy binary FxCop based static analysis.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6ead130f-1feb-2367-ef07-9f73ec63b96f
* Version Independent ID: 1900bccb-13aa-120d-81cb-91e43702a141
* Content: [SuppressMessageAttribute.Scope Property (System.Diagnostics.CodeAnalysis)](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute.scope?view=netframework-4.7.2)
* Content Source: [xml/System.Diagnostics.CodeAnalysis/SuppressMessageAttribute.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Diagnostics.CodeAnalysis/SuppressMessageAttribute.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by mavasani** [comment]
Tagging @gewarren

**Comment by gewarren** [comment]
@gregvanl You are the API ref expert...what's the best way to go about adding the new scope value of "NamespaceAndChildren" to the docs?

**Comment by mavasani** [comment]
@gregvanl Note that there is a subtle design here. `Scope` is just a string, whose allowed values are defined in the attribute API. The legacy FxCop infrastructure allowed the current values, and so when we added SuppressMessageAttribute encoding/decoding support to Roslyn compilers, we supported these allowed values. The new scope value's encoding/decoding is only added to Roslyn compilers, and legacy FxCop will not understand this scope. 

In short, I am not sure if the allowed values or their description can directly come from the Attribute's xml doc comments, as the allowed values are open ended and for the individual encoder/decoder's to support. I am not sure how such cases are handled in our documentation.

**Comment by mairaw** [comment]
@gregvanl there’s no auto-generated content for the .NET reference. You can just click edit and add the text you want.

**Comment by mavasani** [comment]
Also note that we are still considering a change to the name for the new suppression scope: https://github.com/dotnet/roslyn/issues/31196, this will be discussed at a design meeting next week. I will post an update here if the chosen name changes.
